Allow kamon metrics to be enable/disable as per need.

parekuti opened the pull request.

velvia commented on cassandra/src/main/scala/filodb.cassandra/columnstore/ChunkRowMapTable.scala:
```diff
@@ -121,16 +122,19 @@ sealed class ChunkRowMapTable(dataset: DatasetRef, connector: FiloCassandraConne
-                    stats: ColumnStoreStats): Future[Response] = {
-    stats.addIndexWriteStats(chunkIds.capacity.toLong + rowNums.capacity.toLong + 4L)
+                    stats: Option[ColumnStoreStats]): Future[Response] = {
+    if(stats.nonEmpty) {
```
space between if and (

velvia commented on cassandra/src/main/scala/filodb.cassandra/columnstore/ChunkTable.scala:
```diff
@@ -62,7 +62,9 @@ sealed class ChunkTable(dataset: DatasetRef, connector: FiloCassandraConnector)
                           id: java.lang.Integer, columnName, bytes)
     }.toSeq
-    stats.addChunkWriteStats(statements.length, chunkBytes)
+    if(stats.nonEmpty) {
```
space

velvia commented on coordinator/src/main/scala/filodb.coordinator/CoordinatorSetup.scala:
```diff
@@ -84,7 +84,10 @@ trait CoordinatorSetup {
     threadPool.shutdown()
-    kamon.Kamon.shutdown()
+    val kamonEnabled = config.getBoolean("kamon-metrics-flag-enabled")
+    if(kamonEnabled) {
```
space

velvia commented on coordinator/src/main/scala/filodb.coordinator/DatasetCoordinatorActor.scala:
```diff
@@ -138,16 +138,43 @@ private[filodb] class DatasetCoordinatorActor(projection: RichProjection,
-    Math.max(flushingRows, 0).toLong
+  val kamonEnabled = config.getBoolean("kamon-metrics-flag-enabled")
+  private val kamonTags = {
+    if(kamonEnabled) {
```
space space space

velvia commented on coordinator/src/main/scala/filodb.coordinator/DatasetCoordinatorActor.scala:
```diff
@@ -194,7 +221,9 @@ private[filodb] class DatasetCoordinatorActor(projection: RichProjection,
     rowsIngested += tempRows.length
-    kamonRowsIngested.increment(tempRows.length)
+    if (kamonEnabled) {
+      kamonRowsIngested.get.increment(tempRows.length)
```
Just do `kamonRowsIngested.foreach(_.increment(...))`

velvia commented on core/src/main/scala/filodb.core/store/ColumnStore.scala:
```diff
@@ -214,23 +218,27 @@ trait CachedMergingColumnStore extends ColumnStore with ColumnStoreScanner with
+                                     version: Int): Future[Response] = Tracer.withNewContext("append-segment") {
     val ctx = Tracer.currentContext
-    stats.segmentAppend()
+    if(stats.nonEmpty) {
```
space between if and (

velvia commented:
@parekuti oh, can you also remove all of the .idea file changes from the PR?  Thanks.

parekuti commented:
@velvia Thank you for your time and comments.

1. Added extra space for if else blocks.
2. Removed .idea files
3. Yes, agree this is a temporary fix for us to get going until we fix the kamon metrics issue.
4. It's not just ColumnStoreStats that i should create kamon-less classes, there are other objects/classes that i should modify to completely get away with Option[].  Let me know if all objects required to be changed.

velvia commented on core/src/main/scala/filodb.core/store/ColumnStore.scala:
```diff
@@ -271,9 +271,9 @@ trait CachedMergingColumnStore extends ColumnStore with ColumnStoreScanner with
                     segment: Segment,
                     version: Int): Future[Response] = {
-    if(stats.isEmpty){
+    if (stats.isEmpty){
```
Also space between ) and {

velvia commented:
@parekuti this is really old, and I think it's fine to squash and merge it now.
